datajunction-query/djqs/engine.py

```diff
@@ -69,20 +68,28 @@ def get_columns_from_description(
 def run_query(
     session: Session,
     query: Query,
+    headers: Optional[Dict[str, str]] = None,
 ) -> List[Tuple[str, List[ColumnMetadata], Stream]]:
     """
     Run a query and return its results.
 
     For each statement we return a tuple with the statement SQL, a description of the
     columns (name and type) and a stream of rows (tuples).
     """
+
     _logger.info("Running query on catalog %s", query.catalog_name)
+
     engine = session.exec(
         select(Engine)
         .where(Engine.name == query.engine_name)
         .where(Engine.version == query.engine_version),
     ).one()
-    if engine.type == EngineType.DUCKDB:
+
+    query_server = headers.get("QUERY_ENGINE_SERVER")
```

Comment:
I think `QUERY_ENGINE_SERVER` should be renamed to something more explicit like `SQLALCHEMY_URI`.